compiler/AST/AggregateType.cpp

```diff
@@ -1618,78 +1618,65 @@ bool AggregateType::addSuperArgs(FnSymbol*                    fn,
       dispatchParents.n            >      0 &&
       symbol->hasFlag(FLAG_EXTERN) == false) {
     if (AggregateType* parent = dispatchParents.v[0]) {
-      if (parent->initializerStyle != DEFINES_CONSTRUCTOR) {
-        CallExpr* superPortion = new CallExpr(".",
-                                              new SymExpr(fn->_this),
-                                              new_CStringSymbol("super"));
-
-        SymExpr*  initPortion  = new SymExpr(new_CStringSymbol("init"));
-        CallExpr* base         = new CallExpr(".", superPortion, initPortion);
-        CallExpr* superCall    = new CallExpr(base);
-
-        if (parent->initializerStyle == DEFINES_NONE_USE_DEFAULT) {
-          // We want to call the compiler-generated all-fields initializer
-
-          // First, ensure we have a default initializer for the parent
-          if (parent->defaultInitializer == NULL) {
-            // ... but only if it is valid to do so
-            if (parent->wantsDefaultInitializer() == true) {
-              parent->buildDefaultInitializer();
-            }
+      CallExpr* superPortion = new CallExpr(".",
+                                            new SymExpr(fn->_this),
+                                            new_CStringSymbol("super"));
+
+      SymExpr*  initPortion  = new SymExpr(new_CStringSymbol("init"));
+      CallExpr* base         = new CallExpr(".", superPortion, initPortion);
+      CallExpr* superCall    = new CallExpr(base);
+
+      if (parent->initializerStyle == DEFINES_NONE_USE_DEFAULT) {
```

Comment:
We might want to consider transforming this into a bool instead of an enum now that there are only two states